src/coupled_driver.cpp

```diff
@@ -242,6 +255,15 @@ void CoupledDriver::execute()
       update_heat_source(i_timestep_ > 0 || i_picard_ > 0);
 
       if (heat.active()) {
+#ifdef _OPENMP
+        omp_set_num_threads(heat.num_threads);
```

Comment:
What happens if `num_threads` wasn't set for the heat driver? It looks like we don't do anything for Nek5000 or the heat surrogate driver right now.

Reply:
`num_threads` is a data member of the base `Driver` class, and it's initialized as -1 as a kind of null value.  But I didn't anticipate that `omp_set_num_threads(-1)` would be called when coupled to Nek5000 or the surrogate, as you correctly pointed out.  Looks like the behavior of setting to a non-positive integer is implementation defined:  https://www.openmp.org/spec-html/5.1/openmpsu120.html#x159-1920003.2.1  

Would it be more reasonable to initialize `num_threads` to 1?

Reply:
I think it would make more sense to call `omp_get_num_threads` at the very start of execution and use that as the default.

Reply:
In the `Driver` constructor, I now set `num_threads` by calling `omp_num_threads`.  The constructors of the derived classes update the value where needed.